scripts/module/keybinder.nut

```diff
@@ -0,0 +1,96 @@
+keys <- {
+    "backspace" : 0x08,
+    "return" : 0x0D,
+    "lshift" : 0xA0,
+    "rshift" : 0xA1,
+    "lctrl" : 0xA2,
+    "rctrl" : 0xA3
+};
+
+function onScriptLoad() {
+  binds <- array(100, null);
+  local it;
+  for(it = 0; it < 10; ++it) {
+    keys.rawset(format("%i", it), it+0x30);
+  }
+  for(it = 0; it < 26; ++it) {
+    keys.rawset(format("%c", it+0x41), it+0x41);
+  }
+  for(it = 0; it < 10; ++it) {
+    keys.rawset(format("num%i", it), it+0x60);
+  }
+  for(it = 1; it < 13; ++it) {
+    keys.rawset(format("f%i", it), it+0x6F);
+  }
+}
+
+function partition( input, separator ) {
+  local ret = array(2, null);
+  local firstSpace = input.find(separator);
+  if (!firstSpace) {
+    firstSpace = input.len()-1;
+  } else {
+    ret[1] = input.slice(firstSpace+1);
+  }
```

Comment:
Wouldn't this return `["foo", null]` for  "foo" with "bar" separator? Why not ["foo", ""] to keep away edge cases?

Reply:
You are right. My trail of thought was that the used functions return `null` when unsuccessful and it felt to me like the most intuitive way to go would be to continue with that tradition.